test/usersTest.js

```diff
@@ -213,11 +220,51 @@ describe('/users POST', () => {
               user.should.be.present;
               user.name.should.be.equal('Franco');
               user.password.should.not.be.equal('123456789');
+              dictum.chai(res);
+              done();
             });
-            dictum.chai(res);
-            done();
           });
         });
     });
   });
 });
+
+describe('/users GET', () => {
+  let request;
+  beforeEach(done => {
+    request = chai.request(server).get('/users');
+    done();
+  });
+
+  it(`should fail because ${sessionManager.HEADER_NAME} header is not being sent`, done => {
+    User.create({
+      name: 'Franco',
+      surname: 'Coronel',
+      email: 'franco.coronel@wolox.com.ar',
+      password: 'passwordFC'
+    });
+    request.catch(err => err.should.have.status(401));
+    done();
+  });
+
+  it('should return all users', done => {
+    User.create({
+      name: 'Franco',
+      surname: 'Coronel',
+      email: 'franco.coronel@wolox.com.ar',
+      password: 'passwordFC'
+    }).then(
+      successfullLogin().then(loginRes => {
+        request.set(sessionManager.HEADER_NAME, loginRes.headers[sessionManager.HEADER_NAME]).then(res => {
+          res.should.have.status(200);
+          res.should.be.json;
+          res.body[0].should.have.property('name');
+          res.body[0].should.have.property('surname');
+          res.body[0].should.have.property('email');
+          dictum.chai(res);
+        });
+      }),
+      done()
```

Comment:
Remember to put the done() inside the then, otherwise you are not waiting to the should statements to be executed.